### PR TITLE
chore: remove react-markdown

### PR DESCRIPTION
**Pull request by snitin315**
Resolve https://github.com/opencollective/opencollective/issues/3713

Requires https://github.com/opencollective/opencollective-api/pull/4919

# Description

- [x] Display policies with `HTMLContent`
- [x] In settings, use a `RichTextEditor` (with `version="simplified"`) for these fields
- [x] Remove `react-markdown`

# Screenshots

Screenshot at 2020-11-25 17-47-48

### CODE REVIEW

File: components/edit-collective/sections/Policies.js [hunk 1]
```diff
@@ -14,17 +14,14 @@ import { parseToBoolean } from '../../../lib/utils';
...
 import StyledButton from '../../StyledButton';
 import StyledInputField from '../../StyledInputField';
 import StyledSelect from '../../StyledSelect';
-import StyledTextarea from '../../StyledTextarea';
 import { H3, P } from '../../Text';
 
 import { getSettingsQuery } from './EditCollectivePage';
 
-const EXPENSE_POLICY_MAX_LENGTH = 16000; // max in database is ~15,500
-const CONTRIBUTION_POLICY_MAX_LENGTH = 3000; // 600 words * 5 characters average length word
```
Comment (znarf) [inline components/edit-collective/sections/Policies.js:26]: This is something that was recently introduced and I think we want to keep it.

Comment (snitin315) [reply components/edit-collective/sections/Policies.js:26]: Sure, I will add `showCount` prop to RichTextEditor 👍🏻 .

**Comment by snitin315** [comment]
# Latest Screenshot

Screenshot at 2020-11-26 14-56-42

File: components/edit-collective/sections/Policies.js (hunk 1 above)
Comment (snitin315) [reply components/edit-collective/sections/Policies.js:26]: fixed

File: components/edit-collective/sections/Policies.js [hunk 2]
```diff
@@ -14,16 +14,15 @@ import { parseToBoolean } from '../../../lib/utils';
...
+const MAX_LENGTH = 50000; // max in database is ~15,500 without html
```
Comment (Betree) [inline components/edit-collective/sections/Policies.js:25]: 50000 is the DB limit for the row content. For the characters count itself we should keep the existing ones

File: components/RichTextEditor.js [hunk 3]
```diff
@@ -419,12 +427,29 @@ export default class RichTextEditor extends React.Component {
...
+            <Container position="relative">
+              <trix-editor
+                ref={this.editorRef}
+                input={this.state.id}
+                autofocus={autoFocus ? true : undefined}
+                placeholder={placeholder}
+              />
+              <Container position="absolute" bottom="1em" right="1em">
+                <StyledTag textTransform="uppercase">
+                  <span>{defaultValue.length}</span>
```
Comment (Betree) [inline components/RichTextEditor.js:447]: Good idea to display that here. However, the way we compute this length is misleading: if I type 10 characters in bold it should display `10` (the length without HTML).

Comment (snitin315) [reply components/RichTextEditor.js:447]: So, do we have any util to strip html?

Comment (snitin315) [reply components/RichTextEditor.js]: I just added one

File: lib/utils.js [hunk 4]
```diff
@@ -212,3 +212,9 @@ export const flattenObjectDeep = obj =>
     (acc, k) => (typeof obj[k] === 'object' ? { ...acc, ...flattenObjectDeep(obj[k]) } : { ...acc, [k]: obj[k] }),
     {},
   );
+
+export const stripHTML = HTMLContent => {
```
Comment (Betree) [inline lib/utils.js:216]: Camelcase

```suggestion
export const stripHTML = htmlContent => {
```

File: lib/utils.js [hunk 5]
```diff
@@ -212,3 +212,9 @@ export const flattenObjectDeep = obj =>
...
+  const regex = /(&nbsp;|<([^>]+)>)/gi;
```
Comment (Betree) [inline lib/utils.js:217]: Let's rather use the `sanitize-html` package that's already installed in the frontend. It should work with something like:

```es6
return sanitizeHtml(content, { allowedTags: [], allowedAttributes: {} })
```

**Review by Betree — COMMENTED** [review]
- The other instance of this component, the "Expenses and Payouts" menu in host settings, needs to be updated too:

image

- If we go over the limit, it's still possible to input text and there's no validation to check the length, we can just save like that. Could we either block the input or add a validation to show an error message (whatever is the simplest) if it's too long?

image

**Comment by snitin315** [comment]
## Validation Error

Screenshot at 2020-12-04 10-04-46

## Expenses & Payouts

Screenshot at 2020-12-04 10-58-19

**Review by Betree — APPROVED** [review]
All good!

**Comment by snitin315** [comment]
I will rebase.

**Comment by snitin315** [comment]
CI is green ✔️ . It should be good to merge now. 

/cc @Betree

**Comment by Betree** [comment]
It is, but I want to wait for Monday to deploy everything at once.

I'll mark the bounty as completed now though, good job!

**Pull request merged**